### PR TITLE
TodoReact

### PR DESCRIPTION
**Pull request by Ali-Barakat**
I couldn't apply style on the app page.. I didn't know why..

### CODE REVIEW

**Comment by Ali-Barakat** [comment]
hope I did will

File: todo-app/src/App.js [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
+  render(){
     return (
       <div>
-        Hello World!
+        <h1>Todo:</h1>
+        <Todo
+          username='Ali'
+          imageType='retro'
+          task='Get out of the bed'
+          TaskDate={new Date(2017, 9, 12).toDateString()}
```
Comment (spirosikmd) [inline todo-app/src/App.js]: By convention props, should be [camel cased](https://www.quora.com/What-is-the-difference-between-Pascal-Case-and-Camel-Case) for example `taskDate`!

File: todo-app/src/todo.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import State from './state.js'
+import Picture from './Picture.js'
+import './todo.css'
+
+export default class Todo extends React.Component{
+    render(){
+        return (
+        <div calssName='todo'>
+            <div calssName='todo--info'>
+                <Picture username={this.props.username} imageType={this.props.imageType} />
```
Comment (spirosikmd) [inline todo-app/src/todo.js]: Hmm why did you add a picture here?

File: todo-app/src/todo.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                <div><Task task={this.props.task} /></div>
+                <div><State state={this.props.state} /></div>
+            <div>
+            <TaskDate TaskDate={this.props.TaskDate} />
```
Comment (spirosikmd) [inline todo-app/src/todo.js]: Again props should be named using camel cased notation `taskDate`!

File: todo-app/src/todo.js [hunk 4]
```diff
@@ -4,21 +4,19 @@ import TaskDate from './TaskDate.js'
 import State from './state.js'
```
Comment (spirosikmd) [inline todo-app/src/todo.js]: The filename should be the same as the component class name, so in this case `Todo.js`. This is just to be consistent with the rest of the project. Other projects might follow other conventions.

File: todo-app/src/todo.js [hunk 5]
```diff
@@ -4,21 +4,19 @@ import TaskDate from './TaskDate.js'
...
 import Picture from './Picture.js'
 import './todo.css'
+import todos from './todos.js'
```
Comment (spirosikmd) [inline todo-app/src/todo.js]: This is not used here, right? You don't need to import it!

File: todo-app/src/state.js [hunk 6]
```diff
@@ -1,11 +1,36 @@
...
+          } return undone
+      }
     render(){
         return (
-        <div>
-            {this.props.state}
-        </div>
+            <div>
+                <input type="checkbox">
+                {/* checked={this.state.checked} */}
```
Comment (spirosikmd) [inline todo-app/src/state.js]: You should remove `{/* checked={this.state.checked} */}` comment to make code easier to read. Also `input` should close like `<input type="checkbox" />`. You don't need `</input>`

File: todo-app/src/state.js [hunk 7]
```diff
@@ -1,11 +1,36 @@
...
-export default class State extends React.Component{
+export default class Status extends React.Component{
+    constructor(props) {
+        super(props);
+        this.state = {
+          done: this.props.status
+        }
+        this.changedStatus = this.changedStatus.bind(this);
+      }
+      changedStatus(){
```
Comment (spirosikmd) [inline todo-app/src/state.js]: This is not used in the file yet? When do you want to call this function? When checkbox input changes right?

File: todo-app/src/state.js [hunk 8]
```diff
@@ -1,11 +1,36 @@
...
+          let changed = !this.state.done
+          this.setState({
+              done: changed
+         })
+
+        }
+      getStatus() {
+          let undone = "Undone"
```
Comment (spirosikmd) [inline todo-app/src/state.js]: `undone` here is not the best name for this variable as it can get both `undone` and `done values.

File: todo-app/src/App.js [hunk 9]
```diff
@@ -1,29 +1,28 @@
...
+            status={todo.status}/>
+          )
+        }): "No Items...."} 
+      </div>
+    )
+  }
+}
+
+
+        {/* <Todo
```
Comment (spirosikmd) [inline todo-app/src/App.js]: You can remove this comment should code is easier to review!

File: todo-app/src/App.js [hunk 10]
```diff
@@ -1,29 +1,28 @@
...
+import todos from './todos.js'
+
 
 export default class App extends React.Component {
   render(){
     return (
       <div>
         <h1>Todo:</h1>
-        <Todo
+        {todos.length !==0 ? todos.map((todo, index) =>{
```
Comment (spirosikmd) [inline todo-app/src/App.js]: Nice conditional rendering!
